boards/arm/nrf5340dk_nrf5340/nrf5340_cpuapp_common.dts

```diff
@@ -12,6 +12,7 @@
 		zephyr,uart-mcumgr = &uart0;
 		zephyr,bt-mon-uart = &uart0;
 		zephyr,bt-c2h-uart = &uart0;
+		zephyr,entropy = &rng_hci;
```

Comment:
does the nrf5340 not have its own entropy generation source?

Reply:
Not accessible to the application core. The `RNG` peripheral is only available to the network core, which runs the Bluetooth controller. AFAIK right now there is no mechanism to get random numbers on the application core, hence this PR.

Reply:
well, the nRF5340 appcore has access to cryptocell and its RNG, which can be chosen as:
```
zephyr,entropy = &cryptocell;
```

@frkv @mbolivar-nordic could one of you take a look here ?

Reply:
Follow up with a little bit more information for this

The app-core has a special relationship to the CryptoCell or any local peripherals on the device. This is built into the mechanism of SPU and the addition of TrustZone security extensions. There is much more security in using this peripheral as-is

There is no need to use IPC communication for the RNG which is much more distanced from security processes and the "Root of Trust" established during boot.

The RNG on the netcore is intended to be used on the radio core, only. What you are providing by exporting this functionality from the netcore to the app-core is an anti-pattern which is not part of our HW design.